src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityAnalyzer.cs

```diff
@@ -173,20 +171,23 @@ SimpleNameSyntax syntax
 					// TODO: not necessarily a good diagnostic for this use-case
 					ctx.ReportDiagnostic( diagnostic );
 				}
-
-
 			}
 		}
 
-		private static (
-		  ImmutableArray<ITypeParameterSymbol> TypeParameters,
-		  ImmutableArray<ITypeSymbol> TypeArguments
-		) GetTypeParamsAndArgs( ISymbol type )
-			=> type switch {
-				IMethodSymbol method => (method.TypeParameters, method.TypeArguments),
-				INamedTypeSymbol namedType => (namedType.TypeParameters, namedType.TypeArguments),
-
-				_ => throw new NotImplementedException(),
-			};
+		private static bool GetTypeParamsAndArgs( ISymbol type, out ImmutableArray<ITypeParameterSymbol> typeParameters, out ImmutableArray<ITypeSymbol> typeArguments ) {
+			switch (type)
+			{
+				case IMethodSymbol method:
+					typeParameters = method.TypeParameters;
+					typeArguments = method.TypeArguments;
+					return true;
+				case INamedTypeSymbol namedType:
+					typeParameters = namedType.TypeParameters;
+					typeArguments = namedType.TypeArguments;
+					return true;
+				default:
+					return false;

```

Comment:
I'm surprised this compiles since the out parameters aren't assigned, but, :shrug:

Reply:
I think it's because they are value types which are always "definitely assigned" (term from spec)